File: LeetCode/sliding window/p3/c++/solution2.cpp
/*  »¬¶¯´°¿Ú°æ  */
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        unordered_map<char,int> window;
        int left = 0,right = 0,maxLen = 0;
        while(right<s.size()){
            int come = s[right++];
            window[come]++;
            while(window[come]>1){
                int go = s[left++];
                window[go]--;
            }
            maxLen = max(maxLen,right-left);
        }
        return maxLen;
    }
};

File: LeetCode/tree/p104/java/solution2.java
class Solution {
    public int maxDepth(TreeNode root) {
        if(root==null) return 0;
        int leftNode = maxDepth(root.left);
        int rightNode = maxDepth(root.right);
        int ans = Math.max(leftNode,rightNode) + 1;
        return ans;
    }
}